setup.py

```diff
@@ -76,8 +93,32 @@
         include_package_data=True,
         extras_require={
             'alldeps': (
-                'numpy >= {}'.format(NUMPY_MIN_VERSION),
-                'scipy >= {}'.format(SCIPY_MIN_VERSION),
+                f'numpy>={NUMPY_MIN_VERSION}',
+                f'scipy>={SCIPY_MIN_VERSION}',
+                f'joblib>={JOBLIB_MIN_VERSION}',
+                f'threadpoolctl>={THREADPOOLCTL_MIN_VERSION}'
+            ),
+            'examples': (
+                f'matplotlib>={MATPLOTLIB_MIN_VERSION}',
+                f'scikit-image>={SCIKIT_IMAGE_MIN_VERSION}',
+                f'pandas>={PANDAS_MIN_VERSION}',
+                f'seaborn>={SEABORN_MIN_VERSION}',
+                f'Pillow>={PILLOW_MIN_VERSION}',
+                f'memory_profiler>={MEMORY_PROFILER_MIN_VERSION}'
```

Comment:
It shouldn't be used in examples, only to build the docs (and should be optional) unless I'm missing something.

Reply:
yes it is required by sphinx-gallery. I don't remember if `sphinx-gallery` raises only a warning or an error if we set to use it in the `conf.py` but it is not installed.

@lucyleeow @lesteve could answer quickly most probably.

Reply:
Yes, but sphinx-gallery is not necessary to run the scripts in the example/ folder, so at least it's not in the right section...

Reply:
It raises a warning.

Reply:
> Yes, but sphinx-gallery is not necessary to run the scripts in the example/ folder, so at least it's not in the right section...

True

> It raises a warning.

So we can consider it optional and omit it.

Reply:
I think `pillow` is also only required for doc building and not required to just run the examples.

Reply:
> Uhm I think that we might need it to run the example with images if I
recall correctly.

I misunderstood then. I thought 'run the scripts in the example/ folder' means e.g., download the `.py` example and running it. You do need it to build the example documentation but I didn't think you need it to just run the code in the example?

Reply:
Pillow isn't used in the examples. I'll move it in the doc section.

Reply:
yes we have `externals/_pilutils.py` to read the images. I was misleading sorry.